app/src/main/java/com/duckduckgo/app/brokensite/api/BrokenSiteSender.kt

```diff
@@ -54,13 +59,22 @@ class BrokenSiteSubmitter @Inject constructor(
     private val appBuildConfig: AppBuildConfig,
     private val dispatcherProvider: DispatcherProvider,
     private val privacyConfig: PrivacyConfig,
+    private val userAllowListRepository: UserAllowListRepository,
+    private val unprotectedTemporary: UnprotectedTemporary,
+    private val contentBlocking: ContentBlocking,
 ) : BrokenSiteSender {
 
     override fun submitBrokenSiteFeedback(brokenSite: BrokenSite) {
-        val isGpcEnabled = (featureToggle.isFeatureEnabled(PrivacyFeatureName.GpcFeatureName.value) && gpc.isEnabled()).toString()
-        val absoluteUrl = Uri.parse(brokenSite.siteUrl).absoluteString
-
         appCoroutineScope.launch(dispatcherProvider.io()) {
+            val isGpcEnabled = (featureToggle.isFeatureEnabled(PrivacyFeatureName.GpcFeatureName.value) && gpc.isEnabled()).toString()
+            val absoluteUrl = Uri.parse(brokenSite.siteUrl).absoluteString
+
+            val domain = brokenSite.siteUrl.toUri().domain()
+            val protectionsState = !userAllowListRepository.isDomainInUserAllowList(domain) &&
+                !unprotectedTemporary.isAnException(brokenSite.siteUrl) &&
+                featureToggle.isFeatureEnabled(PrivacyFeatureName.ContentBlockingFeatureName.value) &&
+                !contentBlocking.isAnException(brokenSite.siteUrl)
```

Comment:
Could you add a few tests for this change?

Reply:
Done!